### ISSUE TITLE
[Bug?]: Redwood 4.0 breaks prerendering with Clerk

### ISSUE THREAD

**Issue by codekrafter** [redwoodjs/redwood#7545]
### What's not working?

When upgrading to Redwood 4.0, prerendering no longer works when using Clerk. This is because the AuthProvider accesses the clerk instance through the window object, which is undefined in a server environment: https://github.com/redwoodjs/redwood/blob/main/packages/auth-providers/clerk/web/src/clerk.tsx#L32

### How do we reproduce the bug?

Setup a project using clerk authentication, and attempt to prerender a page.

### What's your environment? (If it applies)

```shell
System:
    OS: macOS 12.6
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.16.0 - /private/var/folders/9l/yw5y1l652mq300y8y3gs_77w0000gn/T/xfs-22809d6d/node
    Yarn: 3.2.1 - /private/var/folders/9l/yw5y1l652mq300y8y3gs_77w0000gn/T/xfs-22809d6d/yarn
  Databases:
    SQLite: 3.37.0 - /usr/bin/sqlite3
  Browsers:
    Chrome: 109.0.5414.119
    Firefox: 108.0.2
    Safari: 15.6.1
  npmPackages:
    @redwoodjs/auth-clerk-setup: 4.0.1 => 4.0.1 
    @redwoodjs/core: ^4.0.1 => 4.0.1
```

### Are you interested in working on this?

- [X] I'm interested in working on this

**Comment by jtoar** [comment]
Thanks @codekrafter, I've confirmed this one; I get the error:

``` 
Starting prerendering...
⠋ Prerendering /about -> web/dist/about.html

You can use `yarn rw prerender --dry-run` to debug

---------- Error rendering path "/about" ----------
ReferenceError: window is not defined
    at Object.get client [as client] (/Users/dom/prjcts/redwood/redwood-app/node_modules/@redwoodjs/auth-clerk-web/dist/clerk.js:21:7)
    at AuthProvider (/Users/dom/prjcts/redwood/redwood-app/node_modules/@redwoodjs/auth/dist/AuthProvider/AuthProvider.js:50:39)
    at processChild (/Users/dom/prjcts/redwood/redwood-app/node_modules/react-dom/cjs/react-dom-server.node.development.js:3353:14)
    at resolve (/Users/dom/prjcts/redwood/redwood-app/node_modules/react-dom/cjs/react-dom-server.node.development.js:3270:5)
    at ReactDOMServerRenderer.render (/Users/dom/prjcts/redwood/redwood-app/node_modules/react-dom/cjs/react-dom-server.node.development.js:3753:22)
    at ReactDOMServerRenderer.read (/Users/dom/prjcts/redwood/redwood-app/node_modules/react-dom/cjs/react-dom-server.node.development.js:3690:29)
    at Object.renderToString (/Users/dom/prjcts/redwood/redwood-app/node_modules/react-dom/cjs/react-dom-server.node.development.js:4298:27)
    at recursivelyRender (/Users/dom/prjcts/redwood/redwood-app/node_modules/@redwoodjs/prerender/dist/runPrerender.js:85:43)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at runPrerender (/Users/dom/prjcts/redwood/redwood-app/node_modules/@redwoodjs/prerender/dist/runPrerender.js:144:27)
    at Task.task (/Users/dom/prjcts/redwood/redwood-app/node_modules/@redwoodjs/cli/dist/commands/prerenderHandler.js:157:35)
    at Task.run (/Users/dom/prjcts/redwood/redwood-app/node_modules/listr2/dist/index.cjs:978:11)
    at /Users/dom/prjcts/redwood/redwood-app/node_modules/p-map/index.js:57:22
✖ Failed to render "/Users/dom/prjcts/redwood/redwood-app/web/src/pages/AboutPage/AboutPage.tsx"

Running diagnostic checks
✔ Diagnostics checks passed 

Tips:
- This could mean that a library you're using does not support SSR.
- Avoid using `window` in the initial render path through your React components without checks. 
  See https://redwoodjs.com/docs/prerender#prerender-utils
- Avoid prerendering Cells with authenticated queries, by conditionally rendering them.
  See https://redwoodjs.com/docs/prerender#common-warnings--errors

Command failed with exit code 1: yarn rw prerender

```

**Comment by jtoar** [comment]
@codekrafter this was fixed in v4.1.0 which was released today
